### PR TITLE
Fixed chunk border issue

### PR DESCRIPTION
**Pull request by mes258**

### CODE REVIEW

File: client/chunk.js [hunk 1]
```diff
@@ -32,9 +32,11 @@ var Chunk = function (position_, blocks_, colors_) {
   }
 
   this.getRandomSpawnPosition = function(){
-    if(validSpawnLocations == []){
+    // If this is the first call for this chunk, populate the valid spots
+    if(validSpawnLocations.length == 0){
```
Comment (vinport16) [inline client/chunk.js]: we don't want to recalculate this every time... something to fix in the future

Comment (vinport16) [reply client/chunk.js]: (in the case that there really aren't any valid spawn locations)

File: client/chunk.js [hunk 2]
```diff
@@ -227,17 +229,28 @@ var Chunk = function (position_, blocks_, colors_) {
     for (let z = 0; z < blocks.length; z++) {
       for (let y = 0; y < blocks[0].length; y++) {
         for (let x = 0; x < blocks[0][0].length; x++) {
-          if(blocks[z][y][x] != null){
-            if(blocks[z][y][x] > 0 && (blocks[z][y+1] ==  null || blocks[z][y+1][x] == 0) && (blocks[z][y+2] == null || blocks[z][y+2][x] == 0)){
+          if(blocks[z][y][x] != null && blocks[z][y][x] > 0){
```
Comment (vinport16) [inline client/chunk.js]: you _could_ simplify this to just, if(blocks[x][y][z]) because 0 is falsy

File: client/chunk.js [hunk 3]
```diff
@@ -227,17 +229,28 @@ var Chunk = function (position_, blocks_, colors_) {
...
+              }
+            } else if (blocks[z][y+1][x] == 0 && blocks[z][y+2][x] == 0){
+              // Current block is below the top two levels of the chunk. Check the two blocks above it in the chunk.
               validSpawnLocations.push(new THREE.Vector3(z+0.5, y+2.51, x+0.5))
             }
           }
         }
       }
     }
+    console.log(validSpawnLocations)
```
Comment (vinport16) [inline client/chunk.js]: remove

File: client/chunk.js (hunk 1 above)
Comment (vinport16) [reply client/chunk.js]: maybe store a variable like, spawnlocationscalculated = false and then set it to true once you've done the operation once

Comment (mes258) [reply client/chunk.js]: This value is stored in the chunk and will be populated after the frist time it runs for the chunk.

Comment (vinport16) [reply client/chunk.js]: yeah but if there are no valid locations it will still have length == 0

Comment (mes258) [reply client/chunk.js]: Added boolean to track if it needs to be calculated.

**Review by vinport16 — APPROVED** [review]

**Pull request merged**